### DISCUSSION TITLE
Does prisma support Indexes on Expressions?

### DISCUSSION THREAD

**Discussion by chrisjaimes** [prisma/prisma#2497] [Q&A] [answered]
Let's say I have the following index expression (which works for defining complex unique constraints) https://www.postgresql.org/docs/current/indexes-expressional.html
```
create table request (
  from_user int,
  to_user int,
  active boolean default true
);

CREATE UNIQUE INDEX request_index_constraint 
ON request(greatest(from_user,to_user), least(from_user,to_user))
WHERE (active);
```

When generating the schema, there is no definition for this unique constraint. `findOne` method won't be able to target these fields in the `where` obj. Maybe prisma only works for Unique Constraints? and not Index constraints.

**Comment by ryands17** [comment]
Hey @chrisjaimes :wave: 
Yes currently this isn't possible in Prisma as it only supports `findOne` queries on unique fields and not unique constraints.

Could you open a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with the above use case.

As for a workaround, querying via `prisma.raw` should do it.

**Reply by chrisjaimes** [reply, marked as answer]
Sure @ryands17 ! Thanks for responding. Just opened #2504 👍

**Comment by ludralph** [comment]
Hi @chrisjaimes,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
